### PR TITLE
DM-44305: Implement Gaussian Process interpolation over bad pixels

### PR DESCRIPTION
**Pull request by PFLeget**
See this ticket --> https://rubinobs.atlassian.net/browse/DM-44305

### CODE REVIEW

File: tests/test_isrFunctions.py [hunk 1]
```diff
@@ -110,11 +110,21 @@ def test_interpolateFromMask(self):
...
+                interpMaskedImage = ipIsr.interpolateFromMask(self.mi, 2.0,
+                                                              growSaturatedFootprints=growFootprints,
+                                                              maskNameList=['SAT'],
+                                                              useLegacyInterp=useLegacyInterp)
+                numBit = ipIsr.countMaskedPixels(interpMaskedImage, "INTRP")
+                if useLegacyInterp:
+                    self.assertEqual(numBit, 40800,
+                                     msg=f"interpolateFromMask with growFootprints={growFootprints}")
+                elif growFootprints == 0 and not useLegacyInterp:
+                    self.assertEqual(numBit, 0,
```
Comment (czwa) [inline tests/test_isrFunctions.py:123]: I'm not sure this makes sense to me (it's entirely possible it's a bug elsewhere).  With growFootprints=0 and using the new interpolator returns 0 interpolated pixels?  The fact that the old interpolator returns the same number for all values also doesn't make sense to me.  If you think it's doing the right thing in this case, that's fine, but I don't think I'm following what's happening here.

**Review by czwa — APPROVED** [review]
No major concerns.  The docstring needs updating, but that's trivial enough.

File: tests/test_isrFunctions.py (hunk 1 above)
Comment (PFLeget) [reply tests/test_isrFunctions.py:124]: In this specific test, the full image is saturated and there is no growFootprints to use. So there is no information to use to interpolate over. On the GP algorithm side I took the choice of not doing anything in that specific case. Let me know if it sounds good or if I should do something else in that specific case.

Comment (czwa) [reply tests/test_isrFunctions.py:124]: I think this is fine as written, it's just been a while since I thought through these tests.

**Pull request merged**